modelzoo/DBMTL/train.py

```diff
@@ -0,0 +1,900 @@
+        save_incremental_checkpoint_secs=args.incremental_ckpt)`
+    '''
+    if args.incremental_ckpt and not args.tf:
+        print("Incremental_Checkpoint is not really enabled.")
+        print("Please see the comments in the code.")
+        sys.exit()
+    
+    with tf.train.MonitoredTrainingSession(
+            master=server.target if server else '',
+            is_chief=tf_config['is_chief'] if tf_config else True,
+            hooks=hooks,
+            scaffold=scaffold,
+            checkpoint_dir=checkpoint_dir,
+            save_checkpoint_steps=save_steps,
+            summary_dir=checkpoint_dir,
+            save_summaries_steps=args.save_steps,
+            config=sess_config) as sess:
+        while not sess.should_stop():
+            sess.run([model.loss, model.train_op])
+    print("Training completed.")
+
+def eval(sess_config, input_hooks, model, data_init_op, steps, checkpoint_dir):
+    model.is_training = False
+    hooks = []
+    hooks.extend(input_hooks)
+
+    scaffold = tf.train.Scaffold(
+        local_init_op=tf.group(tf.local_variables_initializer(), data_init_op))
+    session_creator = tf.train.ChiefSessionCreator(
+        scaffold=scaffold, checkpoint_dir=checkpoint_dir, config=sess_config)
+    writer = tf.summary.FileWriter(os.path.join(checkpoint_dir, 'eval'))
+    merged = tf.summary.merge_all()
+
+    with tf.train.MonitoredSession(session_creator=session_creator,
+                                   hooks=hooks) as sess:
+        for _in in range(1, steps + 1):
+            if (_in != steps):
+                sess.run([model.acc_op, model.auc_op])
+                if (_in % 1000 == 0):
+                    print("Evaluation complete:[{}/{}]".format(_in, steps))
+            else:
+                eval_acc, eval_auc, events = sess.run(
+                    [model.acc_op, model.auc_op, merged])
+                writer.add_summary(events, _in)
+                print("Evaluation complete:[{}/{}]".format(_in, steps))
+                print("ACC = {}\nAUC = {}".format(eval_acc, eval_auc))
+
+def main(tf_config=None, server=None):
+    # check dataset and count data set size
+    print("Checking dataset...")
+    train_file = os.path.join(args.data_location, 'train')
+    test_file = os.path.join(args.data_location, 'eval')
+
+    if (not os.path.exists(train_file)) or (not os.path.exists(test_file)):
+        print("Dataset does not exist in the given data_location.")
+        sys.exit()
+    no_of_training_examples = sum(1 for line in open(train_file))
+    no_of_test_examples = sum(1 for line in open(test_file))
+    print("Number of training dataset is {}".format(no_of_training_examples))
+    print("Number of test dataset is {}".format(no_of_test_examples))
+
+    # set batch size, epoch & steps
+    batch_size = math.ceil(
+        args.batch_size / args.micro_batch
+    ) if args.micro_batch and not args.tf else args.batch_size
+
+    if args.steps == 0:
+        no_of_epochs = 3
```

Comment:
Please be sure that by python train.py, the model will be trained for at least 3000 steps. Because the test script in DeepRec will run python train.py and collect the gsteps between 2000~3000. The script is under DeepRec/tests/model_benchmark.

Reply:
Changed

Reply:
Please change deafult steps by `no_of_epochs`